docs/testplan/GCU-Dynamic-ACL-testplan.md

```diff
@@ -0,0 +1,438 @@
+# Dynamic ACL Update via GCU Test Plan
+
+## Overview
+
+This test plan will certify that Generic Config Updater (GCU) is able to properly add, remove, and update ACL Table Types, ACL Tables, and ACL Rules.
+
+## Testbed
+
+The test will run on T0 testbeds.
+
+## Setup Configuration
+
+No setup pre-configuration is required, the test will configure and return the testbed to its original state.
+
+## Testing Plan
+
+To test the capability of GCU to dynamically update ACLs, we will utilize various Json Patch files to create, update, and remove various ACL Tables and Rules.  The contents of the Json Patch files, as well as additional details about verification processes, will be defined in the last section of this document, **JSON Patch Files and Expected Results**.
+
+### Test Case # 1 - Create and apply custom ACL table without rules
+
+#### Test Objective
+
+Verify that we can utilize GCU to create a custom ACL Table Type, and then create an ACL Table from this type
+
+#### Testing Steps
+
+- Create a new ACL Table Type
+
+- Create an ACL Table utilizing this ACL Table Type
+
+- Verify that both operations were successful
+
+- Verify that output of "show acl table {tablename}" matches expected output
+
+### Test Case # 2 - Create a drop rule within custom table
+
+#### Test Objective
+
+Verify that we can create a single drop rule utilizing GCU
+
+#### Testing Steps
+
+- Create a new ACL Table Type
+
+- Create an ACL Table utilizing this ACL Table Type
+
+- Create a new drop rule on our new ACL Table
+
+- Verify that all operations were successful
+
+- Verify that output of "show acl rule | grep {rule_name}" matches expected output
+
+### Test Case # 3 - Create 2 forward rules witihin custom ACL table
+
+#### Test Objective
+
+Verify that we can create a forward rule utilizing GCU, and that we can create forward rules for both ipv4 and ipv6
+
+#### Testing Steps
+
+- Create a new ACL Table Type
+
+- Create an ACL Table utilizing this ACL Table Type
+
+- Create 2 new forwarding rules on our new ACL Table
+
+- Verify that all operations were successful
+
+- Verify that for both rules created, "show acl rule | grep {rulename}" matches expected output for both rules
+
+### Test Case # 4 - Remove a drop rule from the ACL table
+
+#### Test Objective
+
+Verify that we can remove a previously created drop rule from our ACL Table
+
+#### Testing Steps
+
+- Create a new ACL Table Type
+
+- Create an ACL Table utilizing this ACL Table Type
+
+- Create a drop rule on this ACL Table
+
+- Remove the drop rule from this ACL Table
+
+- Verify that all operations were successful
+
+- Verify that the result of "show acl rule {rule_name}" has no relevant output
+
```

Comment:
Same as above, please send some packet to ensure the port is not blocked after removing the dropping rule.